File: serve.js
#!/usr/bin/env node
/**
 * Simple Node.js HTTP server for GitHub User Finder development
 * Serves the application with proper CORS headers for ES6 modules
 */

const http = require('http');
const fs = require('fs');
const path = require('path');
const url = require('url');

const MIME_TYPES = {
    '.html': 'text/html',
    '.js': 'application/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.ico': 'image/x-icon'
};

function getMimeType(filePath) {
    const ext = path.extname(filePath).toLowerCase();
    return MIME_TYPES[ext] || 'application/octet-stream';
}

function createServer() {
    return http.createServer((req, res) => {
        // Add CORS headers
        res.setHeader('Access-Control-Allow-Origin', '*');
        res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
        res.setHeader('Access-Control-Allow-Headers', '*');

        if (req.method === 'OPTIONS') {
            res.writeHead(200);
            res.end();
            return;
        }

        const parsedUrl = url.parse(req.url);
        let pathname = parsedUrl.pathname;

        // Default to index.html for root
        if (pathname === '/') {
            pathname = '/index.html';
        }

        const filePath = path.join(__dirname, pathname);

        fs.readFile(filePath, (err, data) => {
            if (err) {
                res.writeHead(404, { 'Content-Type': 'text/plain' });
                res.end('404 Not Found');
                return;
            }

            const mimeType = getMimeType(filePath);
            res.writeHead(200, { 'Content-Type': mimeType });
            res.end(data);
        });
    });
}

function main() {
    const port = process.argv[2] || 8000;
    const server = createServer();

    server.listen(port, () => {
        console.log('🚀 GitHub User Finder development server');
        console.log(`📡 Serving at http://localhost:${port}`);
        console.log(`📁 Directory: ${__dirname}`);
        console.log('🔧 ES6 modules enabled with CORS support');
        console.log('⏹️  Press Ctrl+C to stop');
    });

    process.on('SIGINT', () => {
        console.log('\n🛑 Server stopped');
        process.exit(0);
    });
}

if (require.main === module) {
    main();
}

File: serve.py
#!/usr/bin/env python3
"""
Simple HTTP server for GitHub User Finder development
Serves the application with proper CORS headers for ES6 modules
"""

import http.server
import socketserver
import os
import sys
from pathlib import Path

class CORSRequestHandler(http.server.SimpleHTTPRequestHandler):
    def end_headers(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', '*')
        super().end_headers()

    def guess_type(self, path):
        mimetype, encoding = super().guess_type(path)
        # Ensure JavaScript modules are served with correct MIME type
        if path.endswith('.js'):
            return 'application/javascript'
        return mimetype

def main():
    port = int(sys.argv[1]) if len(sys.argv) > 1 else 8000
    
    # Change to the directory containing this script
    script_dir = Path(__file__).parent
    os.chdir(script_dir)
    
    with socketserver.TCPServer(("", port), CORSRequestHandler) as httpd:
        print(f"🚀 GitHub User Finder development server")
        print(f"📡 Serving at http://localhost:{port}")
        print(f"📁 Directory: {script_dir}")
        print(f"🔧 ES6 modules enabled with CORS support")
        print(f"⏹️  Press Ctrl+C to stop")
        
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print(f"\n🛑 Server stopped")

if __name__ == "__main__":
    main()
